Support swift v3 authentication

icewind1991 opened the pull request:
- [x] requires https://github.com/nextcloud/server/pull/8259
- [x] requires https://github.com/nextcloud/server/pull/8314

Add support for using swift v3 authentication for external and primary storage

cc @oparoz

icewind1991 committed:
add support for v3 swift auth

Signed-off-by: Robin Appelman <robin@icewind.nl>

MorrisJobke commented:
Conflicts :/

rullzer commented:
Rebased on master

rullzer commented:
@icewind1991 can we please have two CI steps one that tests v2 and one that tests v3.

Also we can probably log a warning if users use V2 right since that has been deprecated since ages?

rullzer committed:
Add v3.0 auth run

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

rullzer committed:
Other config

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

rullzer commented:
A lot of my changes can be removed or squased later. It looks like we need a bit smarter way to detect how to find out v2 or v3 is used

rullzer committed:
User proper username + pass

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

rullzer committed:
Use username + pass in config

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

rullzer commented:
I *think* (tm) this works now. I'm not sure if we maybe should make it explicit which version is used in the config?

I also want CI to pass on this (but the one node running seems to not like swift). But I really don't want to break stuff here (we did that before with object storages so please not again).

rullzer committed:
Some swift v3 example config

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

rullzer commented:
So it seems for some dark reason on some drone hosts this does :boom: on others it works as expected.

rullzer committed:
Kill objectstore CI if we can't create a file

This is more of a hack. But one of the nodes won't properly run this. No
sense in waiting 60 minutes

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

rullzer commented:
One succesfull v3 run! https://drone.nextcloud.com/nextcloud/server/5122/248

icewind1991 committed:
adjust external storage config format to the one used in primary storage

Signed-off-by: Robin Appelman <robin@icewind.nl>

oparoz commented on config/config.sample.php:
```diff
@@ -1214,6 +1214,28 @@
+			'password' => 'swift',
+			'domain' => [
+				'name' => 'default',
+			]
```
@rullzer missing comma?

MorrisJobke replied:
This is optional for the last element of an array (we usually add it to make the possible diff better readable.

scysys commented:
Merged this in my installation and this solved v3 issues on my side too!
